File: 0029-divide-two-integers/0029-divide-two-integers.py
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        h = 2 ** 31 - 1
        l = -2 ** 31

        s1 = -1 if dividend < 0 else 1
        s2 = -1 if divisor < 0 else 1

        dividend = abs(dividend)
        divisor = abs(divisor)
        res = len(range(0, dividend - divisor + 1, divisor))
        if s1 == -1 or s2 == -1:
            res = -res
        if s1 == s2:
            res = abs(res)

        return max(l, min(res, h))

File: 0029-divide-two-integers/0029-divide-two-integers.go
func divide(dividend int, divisor int) int {
    var max int = math.MaxInt32
    var min int = math.MinInt32
    
    if (dividend == min) && (divisor == -1) {
        return max
    }
    if ((dividend == max) && (divisor == max)) ||
        ((dividend == min) && (divisor == min)) {
        return 1
    }
    if (dividend == min) && (divisor == max) {
        return -1
    }
    if (dividend == max && divisor == min) {
        return 0
    }
    return dividend/divisor
}